dateparser/date.py

```diff
@@ -423,7 +423,7 @@ def get_date_data(self, date_string, date_formats=None):
             if parsed_date:
                 parsed_date['locale'] = locale.shortname
                 if self.try_previous_locales:
-                    self.previous_locales.add(locale)
+                    self.previous_locales.update({locale: locale})
```

Comment:
```suggestion
                    self.previous_locales[locale] = None
```

I think adding `None` as value makes sense here as we only need the keys. On the other hand, assigning the value with `[]` is about three times faster than using `.update`, so I would prefer to use it. :)